### ISSUE TITLE
[Visual Requirement-Help-References]Luminosity ratio is less than 4.5:1 for 'show leds' on Basic pane

### ISSUE THREAD

**Issue by v-mesala** [Microsoft/pxt#2605]
**User Experience**
If the luminosity ratio is less than 4.5:1 for all the text content then it will be difficult for the visually impaired user to see what is present on the screen.

**Test Environment**
OS: RS2 Version 1703(OS Build 15063.483)
Platform: Edge.
Tool: Color contrast Analyzer

**Repro Steps**
1-Navigate to https://makecode.microbit.org/acc
2-Navigate to the Micro bit section element and select Code control given on it.
3-Navigate to the Help control lying in the header section on the page and select it.
4-Navigate to 'References' control and select it.
5-Navigate to any step like Basic and select it.
6-Launch Color Contrast Analyzer tool using WAT tool bar.
8-Verify, the luminosity ratio is greater than 4.5:1 for all text or not.

**Actual Result**
Luminosity ratio is less than 4.5:1 for 'show leds' on Basic pane.
**Expected Result**
Luminosity ratio should be greater than 4.5:1 for 'show leds' on Basic pane.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a73546c7-765f-489c-b18f-afb659fe99e6}

**Suggested fix**
1.1-Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text
1.2- If design allows , Not specify background color, not specify text color, and not using technology features that change those defaults .In this technique the author avoids having to do any contrast measures by simply not specifying the text color and not specifying the background. As a result the colors of both are completely determined by the user agent.
Tips:-
1.The easiest fix is to increase the brightness of light colors and decrease the brightness of darker colors.
2.Exaggerate lightness differences between foreground and background colors, and avoid using colors of similar brightness adjacent to one another, even if they differ in saturation or hue.
3.During design Use the Color Contrast Analyzer to determine whether new colors have color contrast ratio of at least 4.5:1.

**Key References for Fix**
http://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html

Please refer the attachment file
![mas1 4 3_references_contrast ratio is not correct](https://user-images.githubusercontent.com/30434417/28820217-75a7e290-76ce-11e7-9cbf-21d094a5b199.png)

**Comment by veler** [comment]
This is part of Blocky, a third-part component.

**Comment by abchatra** [comment]
We have got exception for Blockly bugs from Julia Liuson (CVP). Closing these as Won't Fix. Following is the reasoning.

----------------------

MakeCode can be used in either Text editor or Block editor.

Blocks editor is inherently visual and is based on google blockly. Usage requires drag and drop functionality and we don’t have a equivalence in MakeCode for this w.r.t accessibility. 
We have evaluated couple of research projects which tries to find a equivalence to block based coding for accessibility needs: 
•	Project Torino (Requires students to buy physical blocks)
•	Accessible blockly (This is text based coding without syntax errors)
Both these are drastic departure from the blockly and provides a complete new experience. These are still in research and we intend not to use them in MakeCode at this point of time.

As workaround, the text editor in MakeCode is the way we see students with the need of accessibility using MakeCode.
